api/metrics.go

```diff
@@ -0,0 +1,30 @@
+// metrics.go
+
+package api
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	// Define your metrics here
+	requestCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "api_requests_total",
+			Help: "Total number of API requests.",
+		},
+		[]string{"method", "endpoint"},
+	)
+)
+
+// Initialize and register metrics
+func init() {
+	prometheus.MustRegister(requestCounter)
+}
+
+// Update the metrics based on the request
+func RecordRequestMetrics(method, endpoint string) {
+	requestCounter.WithLabelValues(method, endpoint).Inc()
+}
```

Comment:
We possibly don't need to export this function from the `api` package, as we don't want to update the associated metric from anywhere else in GatewayD. Unexport the function, as in use camel case.